### PR TITLE
🧪 Scout: verify ICU pound scoping in complex nested structures

### PR DESCRIPTION
**Pull request by cungminh2710**
💡 What: Added TestCountPoundsComplexNesting to invariant_test.go.
🎯 Why: To ensure structural parity checks correctly handle complex ICU nesting and HTML tags, which are prone to scoping errors.
🧩 Scope: internal/i18n/icuparser/invariant.go (logic protected) and invariant_test.go.
✅ Verification: Ran 'go test -v ./internal/i18n/icuparser/...' and workspace-wide 'go test ./...'.
🛡️ Confidence: High. These tests specifically target edge cases in recursive AST traversal and scoping boundaries.

---
*PR created automatically by Jules for task [2063025971716253280](https://jules.google.com/task/2063025971716253280) started by @cungminh2710*

### CODE REVIEW

File: internal/i18n/icuparser/invariant_test.go [hunk 1]
```diff
@@ -419,3 +419,56 @@ func TestParseInvariantIncludesTypedBlocks(t *testing.T) {
...
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			inv, err := ParseInvariant(tt.msg)
+			if err != nil {
+				t.Fatalf("ParseInvariant failed: %v", err)
+			}
+			if !SameICUBlocks(inv.ICUBlocks, tt.want) {
+				t.Errorf("ICUBlocks mismatch\n got: %s\nwant: %s", FormatICUBlocks(inv.ICUBlocks), FormatICUBlocks(tt.want))
+			}
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply internal/i18n/icuparser/invariant_test.go]: @jules fix

File: internal/i18n/icuparser/invariant_test.go [hunk 2]
```diff
@@ -419,3 +419,56 @@ func TestParseInvariantIncludesTypedBlocks(t *testing.T) {
...
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply internal/i18n/icuparser/invariant_test.go:477]: @jules fix
